backend/src/main/java/com/javahelp/backend/data/DynamoDBUserStore.java

```diff
@@ -248,145 +315,62 @@ private static Map<String, AttributeValue> updateFromUser(User u) {
+     * @return whether the map matches the expected fields in a provider
      */
-    private static class SerializableProviderUserInfo implements Serializable, Supplier<UserInfo> {
-
-        String email, practice, address, phoneNumber;
-
-        boolean certified;
-
-        /**
-         * Instantiates a {@link SerializableProviderUserInfo}
-         */
-        public SerializableProviderUserInfo() {
-
-        }
-
-        /**
-         * Constructs a {@link SerializableProviderUserInfo} object
-         *
-         * @param base {@link ProviderUserInfo} to construct from
-         */
-        public SerializableProviderUserInfo(ProviderUserInfo base) {
-            email = base.getEmailAddress();
-            practice = base.getPracticeName();
-            address = base.getAddress();
-            phoneNumber = base.getPhoneNumber();
-            certified = base.isCertified();
-        }
-
-        @Override
-        public UserInfo get() {
-            ProviderUserInfo p = new ProviderUserInfo(email, address, phoneNumber, practice);
-            p.setCertified(certified);
-            return p;
-        }
+    private static boolean matchesProvider(Map<String, AttributeValue> item) {
+        return item.containsKey("infotype") &&
+                "PROVIDER".equals(item.get("infotype").getS()) &&
+                item.containsKey("address") &&
+                item.containsKey("phoneNumber") &&
+                item.containsKey("certified") &&
+                item.containsKey("practiceName");
     }
 
     /**
-     * Serializable wrapper around fields of a {@link ClientUserInfo}
+     * @param item {@link Map} to consider
+     * @return whether the map matches the expected fields in a provider
      */
-    private static class SerializableClientUserInfo implements Serializable, Supplier<UserInfo> {
-
-        String email, first, last, address, phoneNumber;
-
-        /**
-         * Constructs a {@link SerializableClientUserInfo} object
-         *
-         * @param base {@link ClientUserInfo} to construct from
-         */
-        public SerializableClientUserInfo(ClientUserInfo base) {
-            email = base.getEmailAddress();
-            first = base.getFirstName();
-            last = base.getLastName();
-            address = base.getAddress();
-            phoneNumber = base.getPhoneNumber();
-        }
-
-        /**
-         * Instantiates a {@link SerializableClientUserInfo}
-         */
-        public SerializableClientUserInfo() {
-
-        }
-
-        @Override
-        public UserInfo get() {
-            return new ClientUserInfo(email, address, phoneNumber, first, last);
-        }
+    private static boolean matchesClient(Map<String, AttributeValue> item) {
+        return item.containsKey("infotype") &&
+                "CLIENT".equals(item.get("infotype").getS()) &&
+                item.containsKey("address") &&
+                item.containsKey("firstName") &&
```

Comment:
Since we are considering gender attribute, I think we should include gender here, maybe.
Also, same comment for provider info code above line 355.

Reply:
I will add these once #80 is merged!